hazelcast/src/hazelcast/client/network.cpp

```diff
@@ -647,6 +683,108 @@ ClientConnectionManagerImpl::check_client_active()
     }
 }
 
+void
+ClientConnectionManagerImpl::initialize_client_on_cluster(boost::uuids::uuid target_cluster_id)
+{
+    try
+    {
+        {
+            std::lock_guard<std::recursive_mutex> guard(client_state_mutex_);
+
+            if (target_cluster_id != cluster_id_){
+                logger_.log(
+                    hazelcast::logger::level::warning,
+                    (
+                        boost::format(
+                            "Won't send client state to cluster: %1%"
+                            " Because switched to a new cluster: %2%"
+                        ) % target_cluster_id % cluster_id_
+                    ).str()
+                );
+
+                return;
+            }
+        }
+
+        client_.get_hazelcast_client_implementation()->send_state_to_cluster();
+
+        {
+            std::lock_guard<std::recursive_mutex> guard(client_state_mutex_);
+
+            if (target_cluster_id == cluster_id_){
+                if (logger_.enabled(hazelcast::logger::level::fine)){
+                    logger_.log(
+                        hazelcast::logger::level::fine,
+                        (
+                            boost::format(
+                                "Client state is sent to cluster: %1%"
+                            ) % target_cluster_id
+                        ).str()
+                    );
+                }
+
+                client_state_ = client_state::INITIALIZED_ON_CLUSTER;
+                fire_life_cycle_event(lifecycle_event::CLIENT_CONNECTED);
+            }
+            else if (logger_.enabled(hazelcast::logger::level::fine)) {
+                logger_.log(
+                    hazelcast::logger::level::warning,
+                    (
+                        boost::format(
+                            "Cannot set client state to %1%"
+                            " because current cluster id: %2%"
+                            " is different than expected cluster"
+                            " id: %3%"
+                        )
+                    ).str()
+                );
+            }
+        }
+    }
+    catch (const std::exception& e)
+    {
+        // TODO: Check whether this cluster_name
+        // matches with `clusterDiscoveryService.current().getClusterName()`
+        auto cluster_name = client_.get_client_config()
+                                   .get_cluster_name();
+
+        logger_.log(
+            hazelcast::logger::level::warning,
+            (
+                boost::format(
+                    "Failure during sending state to the cluster. %1%"
+                ) % e.what()
+            ).str()
+        );
+
+        {
+            std::lock_guard<std::recursive_mutex> guard(client_state_mutex_);
+
+            if (cluster_id_ == target_cluster_id){
+                if (logger_.enabled(hazelcast::logger::level::fine)){
```

Comment:
is there a specific reason of not using HZ_LOG?

Reply:
Fixed.